specification/keyvault/data-plane/Security.KeyVault.Secrets/routes.tsp

```diff
@@ -0,0 +1,278 @@
+import "@azure-tools/typespec-azure-core";
+import "@azure-tools/typespec-client-generator-core";
+import "@typespec/rest";
+import "./models.tsp";
+
+using Azure.Core;
+using Azure.Core.Traits;
+using Azure.ClientGenerator.Core;
+using TypeSpec.Rest;
+using TypeSpec.Http;
+
+namespace KeyVault;
+
+alias KeyVaultOperations = StandardResourceOperations;
+
+alias KeyVaultListParameters = ListQueryParametersTrait<{
+  /**
+   * Maximum number of results to return in a page. If not specified the service
+   * will return up to 25 results.
+   */
+  @maxValue(25)
+  @minValue(1)
+  @query("maxresults")
+  maxresults?: int32;
+}>;
+
+/**
+ *  The SET operation adds a secret to the Azure Key Vault. If the named secret
+ * already exists, Azure Key Vault creates a new version of that secret. This
+ * operation requires the secrets/set permission.
+ */
+#suppress "@azure-tools/typespec-azure-core/use-standard-operations" "Foundations.Operation is necessary for Key Vault"
+@summary("Sets a secret in a specified key vault.")
+@route("/secrets/{secretName}")
+@put
+op setSecret is Azure.Core.Foundations.Operation<
```

Comment:
This looks like a resource create or replace?

Reply:
See https://github.com/Azure/azure-rest-api-specs/pull/29249#discussion_r1635638503. Many of these operations could be standard resource operations if the service returned a resource name instead of a resource ID in the form of a full URL. I couldn't get operations to conform to using the latter as a resource key, and I'm not aware of a way to override the route otherwise.

Reply:
So the secret name is not returned? It is only in the path of the request?

Reply:
It's not returned as a standalone property, no. It's part of the `id` but needs to be parsed manually -- here's an [example from a recording](https://github.com/Azure/azure-sdk-assets/blob/python/keyvault/azure-keyvault-secrets_ffad208c83/python/sdk/keyvault/azure-keyvault-secrets/tests/recordings/test_samples_secrets.pyTestExamplesKeyVaulttest_example_secret_crud_operations%5B7.6-preview.1%5D.json#L69).
```
"id": "https://vaultname.vault.azure.net/secrets/{secretName}/{secretVersion}"
```